src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -1368,11 +1375,24 @@ public void afterDisconnect(SlaveComputer slaveComputer, final TaskListener list
             srv.shutdown();
         }
 
+        if (tearingDownConnection) {
```

Comment:
Could we simply move the call to `tearDownConnection` into a background thread so that we do not block `afterDisconnect`?

Reply:
That's what I would like to do (see my comment in the PR description).